packages/git/src/browser/git-repository-provider.ts

```diff
@@ -113,7 +113,8 @@ export class GitRepositoryProvider {
 
     async refresh(options?: GitRefreshOptions): Promise<void> {
         const roots: FileStat[] = [];
-        for (const root of await this.workspaceService.roots) {
```

Comment:
What is the difference? more code?

Reply:
no difference from the functionality point of view, but it makes the code easier to unit test, as `tryGetRoots()` can be stubbed in the following way:
```ts
tryGetRoots.onCall(0).returns(one value);
tryGetRoots.onCall(1).returns(another value);
tryGetRoots.onCall(2).returns(and another value);
```
while `await ws.roots` is a getter from WorkspaceService, and 
```ts
roots.onCall(0).value(Promise.resolve(one value))
roots.onCall(1).value(Promise.resolve(another value))
roots.onCall(2).value(Promise.resolve(and another value))
```
did not works - only "and another value" from the 3rd line gets resolved.

Reply:
ok :)